src/cascadia/TerminalControl/TermControlUiaProvider.cpp

```diff
@@ -82,49 +87,73 @@ IFACEMETHODIMP TermControlUiaProvider::get_FragmentRoot(_COM_Outptr_result_maybe
 
 const COORD TermControlUiaProvider::GetFontSize() const
 {
-    return _termControl.GetActualFont().GetSize();
+    return _termControl->GetActualFont().GetSize();
 }
 
 const winrt::Windows::UI::Xaml::Thickness TermControlUiaProvider::GetPadding() const
 {
-    return _termControl.GetPadding();
+    return _termControl->GetPadding();
 }
 
-std::deque<UiaTextRangeBase*> TermControlUiaProvider::GetSelectionRanges(_In_ IRawElementProviderSimple* const pProvider)
+HRESULT TermControlUiaProvider::GetSelectionRanges(_In_ IRawElementProviderSimple* pProvider, _Out_ std::deque<ComPtr<UiaTextRangeBase>>& result)
+try
 {
-    std::deque<UiaTextRangeBase*> result;
+    typename std::remove_reference<decltype(result)>::type temporaryResult;
+
+    std::deque<ComPtr<UiaTextRange>> ranges;
+    RETURN_IF_FAILED(UiaTextRange::GetSelectionRanges(_pData, pProvider, ranges));

```

Comment:
Do you want to ensure that `result` is cleared on the way in such that it contains nothing if you return on failure at this point?